### PR TITLE
Fix memory leak detected by address sanitizer

### PR DESCRIPTION
**Pull request by Adriankhl**
When I was trying to solve the crash in https://github.com/minetest/minetest/issues/10895, I tried to build minetest with `-fsanitize=address` to see if it was caused by some sort of memory leak. Though there are leaks, they are not related to the issue.

The leaks are reported after I close minetest, so their effects are minor and shouldn't affect the game, but I think it is good to fix any leak if possible. 

I have tested that minetest runs fine on my machine with this PR. It is still possible that it affects some sorts of resource destruction so reviews are needed.

### CODE REVIEW

File: src/gui/guiEngine.cpp [hunk 1]
```diff
@@ -75,9 +75,8 @@ video::ITexture *MenuTextureSource::getTexture(const std::string &name, u32 *id)
 	if (name.empty())
 		return NULL;
 
-	m_to_delete.insert(name);
-
 #if ENABLE_GLES
+	m_to_delete.insert(name);
```
Comment (sfan5) [inline src/gui/guiEngine.cpp:79]: This should be moved next to the `addTexture` call it belongs to

**Review by sfan5 — APPROVED** [review]
Tested

File: src/gui/guiEngine.cpp (hunk 1 above)
Comment (Adriankhl) [reply src/gui/guiEngine.cpp]: Done

**Review by SmallJoker — APPROVED** [review]
Game still works, found no issues.

**Pull request merged**